### PR TITLE
Add config option to disable auto-generated usernames

### PR DESCRIPTION
**Pull request by ColdDev**
This pull request implements the ability for server owners to be able to disable "default" Eaglercraft usernames from being used to join servers.

The reason why I am making this pull request is that I have seen instances where these default usernames (especially on servers like Asspixel) take up unnecessary space, i.e. people just playing Eaglercraft for the first time and playing a server once then never again. By adding this feature, server owners will be able to prevent these types of usernames from cluttering auth databases and forcing users to create a unique username, as indicated by the kick message.

Hopefully this PR is satisfactory - let me know if anything needs changing. Thanks! :)

### CODE REVIEW

**Comment by ShoelessTom** [comment]
WAKE UP WAKE UP WAKE UP WAKE UP WAKE UP WAKE UP WAKE UP WAKE
image

File: eaglercraftbungee/src/main/java/net/md_5/bungee/connection/InitialHandler.java [hunk 1]
```diff
@@ -128,6 +128,11 @@ public void handle(final Packet2Handshake handshake) throws Exception {
 			this.disconnect("Go fuck yourself");
 			return;
 		}
+		if (!un.equals(un.replaceAll("\bYeeish|Yee|Yeer|Yeeler|Eagler|Eagl|Darver|Darvler|Vool|Vigg|Deev|Yigg|Yeeg\b", "_").trim())
```
Comment (ColdDev) [inline eaglercraftbungee/src/main/java/net/md_5/bungee/connection/InitialHandler.java:131]: I feel like this regex could be optimized - if you guys have any idea on how I could approach this for the purpose of performance, cleanliness etc. just let me know and I'll be happy to look into it.

Comment (Abnumality) [reply eaglercraftbungee/src/main/java/net/md_5/bungee/connection/InitialHandler.java:131]: Have you considered (🤔) rewriting (📝) this in Rust (🚀)? (So it’s faster)

**Comment by 3kh0** [comment]
They could just have a system that clears player data after inactivity, (eg, a month or so)

**Comment by ColdDev** [comment]
> They could just have a system that clears player data after inactivity, (eg, a month or so)

This is not ideal for most servers, since most server owners are not experienced in plugin development. In addition, servers using Bungee auth would have an even more difficult time implementing this feature. This feature adds an easy way to outright block these usernames rather than having to remove the data of those users after a certain time period has passed.

**Comment by 3kh0** [comment]
@ColdDev I see your point, I forgot to add that this can be added within the Bungee Auth system, and it would only clear data from the auto-generated usernames. And this would be a config in the plugin, not the server. I am kinda a noob at this so please forgive me if I am saying stupid stuff. :)

**Comment by ColdDev** [comment]
> @ColdDev I see your point, I forgot to add that this can be added within the Bungee Auth system, and it would only clear data from the auto-generated usernames. And this would be a config in the plugin, not the server. I am kinda a noob at this so please forgive me if I am saying stupid stuff. :)

No worries! You're making completely valid points, and for some servers, a system like the one you proposed might be more appropriate, however for other servers they might not have the experience to make such a system work.

**Comment by ColdDev** [comment]
@lax1dude (apologies for the mention) Any chance you can take a look at this? I feel like some people might benefit from this

**Pull request closed without merging**